src/sagas/index.js

```diff
@@ -520,23 +510,41 @@ export function* fetchVmSnapshots ({ vmId }) {
 
   if (snapshots && snapshots.snapshot) {
     snapshotsInternal = snapshots.snapshot.map(snapshot => Api.snapshotToInternal({ snapshot }))
-
-    // NOTE: Snapshot Disks and Nics are not currently (Sept-2018) available via
-    //       additional/follow param on the snapshot fetch.  We need to fetch them
-    //       directly.
-    for (const snapshot of snapshotsInternal) {
-      const follows = yield all([
-        call(fetchVmSnapshotDisks, { vmId, snapshotId: snapshot.id }),
-        call(fetchVmSnapshotNics, { vmId, snapshotId: snapshot.id }),
-      ])
-      snapshot.disks = follows[0]
-      snapshot.nics = follows[1]
-    }
+    yield parallelFetchAndPopulateSnapshotDisksAndNics(vmId, snapshotsInternal)
   }
 
   yield put(setVmSnapshots({ vmId, snapshots: snapshotsInternal }))
 }
 
+function* parallelFetchAndPopulateSnapshotDisksAndNics (vmId, snapshots) {
+  if (!Array.isArray(snapshots) || snapshots.length === 0) {
+    return
+  }
+
+  // NOTE 1: Snapshot Disks and Nics are not currently (Sept-2018) available via
+  //         additional/follow param on the VM/snapshot fetch.  We need to fetch them
+  //         directly.
+  // NOTE 2: Looks like the `?follows=snapshots` returns the active and regular
+  //         snapshots.  The active snapshot does not contain links to disks and nics,
+  //         but the regular ones do.  That is probably the reason why the VM call
+  //         `?follows=snapshots.disks` fails.
+  yield all(
```

Comment:
1. `map()` usually has no side effects - here we use it only for side effects (fetching and mutating)
2. `all()` could fail on any request (not sure if it will leave the snapshots partially updated/corrupted or not)

How about sth like (pseudo-code)):
```
const [id2nics, id2disks] = yield all([
   call( all, Object.fromEntries(nonActive.map({id}=> [id, call(fetchVmSnapshotNics,....)]))),
   call( all, Object.fromEntries(nonActive.map({id}=> [id, call(fetchVmSnapshotDisks,....)]))),
])

if(id2nics) {
   snapshots.forEach(s => s.nics = id2nics[s.id] ?? [])
} else {
  //  trigger user notification about nics
}
```
This should give us all-or-nothing updates and a way to notify user about failed operation

Reply:
>   1. `map()` usually has no side effects - here we use it only for side effects (fetching and mutating)
>

`map()` always turns one set of things into another set of things - in this case, one snapshot to one disks fetch plus one nics fetch.  Mapping over things to get a set of actions to call for each thing is a pretty standard form in sagas. 

I'll make a few changes to remove the `map(entity to callback)` to see if I like how it looks better.

> 
>   2. `all()` could fail on any request (not sure if it will leave the snapshots partially updated/corrupted or not)
> 

This is no different than previous.  All that is changing is fetching them all at the same time instead of iterating though each snapshot.  If the call failed before, the VM's snapshot data would be no better or worse than before this PR.

> 
> How about sth like (pseudo-code)):
> 
> ```
> const [id2nics, id2disks] = yield all([
>    call( all, Object.fromEntries(nonActive.map({id}=> [id, call(fetchVmSnapshotNics,....)]))),
>    call( all, Object.fromEntries(nonActive.map({id}=> [id, call(fetchVmSnapshotDisks,....)]))),
> ])
> ```

Oddly, this would not result in the same make-all-fetches-at-once behavior as I was going for.  Only a flat array or flat object would do that.  Annoying for sure.

> ```
> if(id2nics) {
>    snapshots.forEach(s => s.nics = id2nics[s.id] ?? [])
> } else {
>   //  trigger user notification about nics
> }
> ```
> 
> This should give us all-or-nothing updates and a way to notify user about failed operation

It would take more than that ... probably a try/catch/finally clause to really do that right.  Plus, that is a functional change that introduces a new notification.  I don't think we need to do that kind of change now.  I'll add a TODO note about maybe detecting that, and a few other things that are fetched differently from the full VM.

Ideally, the rest endpoint would actually work with the follows call and remove the need to do these calls all together.